tests/Functional/Net/SFTPUserStoryTest.php

```diff
@@ -649,5 +649,22 @@ public function testStatVsLstat($sftp)
         $this->assertSame($stat['type'], NET_SFTP_TYPE_SYMLINK);
 
         $sftp->enableStatCache();
+
+        return $sftp;
+    }
+
+    /**
+     * @depends testStatVsLstat
+     * @group github830
+     */
+    public function testEndlessLoopOnUpload($sftp)
```

Comment:
With "the second hunk" I mean the addition of `$size-= $offset;`, not the code that is already there in the else block.